builtin/credential/aws/path_login.go

```diff
@@ -281,6 +269,17 @@ func (b *backend) pathLoginUpdate(
 	storedIdentity.PendingTime = identityDoc.PendingTime
 	storedIdentity.DisallowReauthentication = disallowReauthentication
 
+	// Performing the clientNonce empty check after determining the DisallowReauthentication
+	// option. This is to make clientNonce optional when DisallowReauthentication is set.
+	if clientNonce == "" && !storedIdentity.DisallowReauthentication {
+		return logical.ErrorResponse("missing nonce"), nil
+	}
+
+	// Limit the lengh to a reasonable length.
+	if len(clientNonce) > 128 && !storedIdentity.DisallowReauthentication {
```

Comment:
I put the check in case the nonce is non-zero and the length is > 128 with DisallowReauthentication set :)
Super paranoia.